### PR TITLE
Coordinates ape5 1to1 framenames

### PR DESCRIPTION
**Pull request by eteq**
This PR addresses an issue that @taldcroft brought up in #28 : There's no really good reason for the feature that `TransformGraph` objects can have many names for each frame class.   In fact, that's probably more confusing than its worth.  So this changes things so that frames have a `name` class attribute that can be set explicitly, or if not, will be taken as `lower()` of the class name.  The functionality for adding names to `TransformGraph` is then removed in favor of using the names on the objects for the `TransformGraph` methods that deal with names.  If it does turn out in the future that aliases really are important in some context, they can always be added back in.

The presence of `name` attribute and the one-to-one assumption it allows then leads to various simplifications in `SkyCoord`... So you'll note this is @taldcroft's favorite kind of PR, with more code removed than added :wink:

Look good to you, @taldcroft ?

### CODE REVIEW

File: astropy/coordinates/sky_coordinate.py [hunk 1]
```diff
@@ -12,32 +12,18 @@
...
-    return dict((cls, name) for name, cls in FRAME_CLASSES().items())
-
-
+# Define a convenience mapping.  This is used like a module constants
+# but is actually dynamically evaluated.
 def FRAME_ATTR_NAMES_SET():
     """Set of all possible frame-specific attributes"""
     out = set()
-    for frame_cls in FRAME_CLASSES().values():
+    for frame_cls in frame_transform_graph._cached_classes:
```
Comment (taldcroft) [inline astropy/coordinates/sky_coordinate.py]: In this context the use of a hidden attribute `_cached_classes` seems confusing.  Shouldn't there be a public attribute or method to access all frame classes in a transform graph?  The label as `cached` also makes me think of something that is temporarily there as a speedup convenience, which I don't think applies here.

**Comment by taldcroft** [comment]
Other than my one minor comment, this is fantastic!  Thanks @eteq.

**Comment by taldcroft** [comment]
Looking through this PR made me realize there are no tests showing that transforming to/from a None-frame skycoord raises an exception.  Maybe that should be added here.

**Comment by eteq** [comment]
The commits I just pushed up blew away the thread above, @taldcroft, so I'll just clarify: I changed `_cached_classes` to `frame_set` as per your suggestion, but the reason it's called "cached" is because it gets regenerated on-demand when the transform graph is altered (see `TransformGraph.invalidate_cache`.  But that's essentially an implementation detail, so you're right that a name like `frame_set` is more sensible now that it's public.

I also added the test you suggested.  With that I think this is good and I will merge this and try to get the main PR actually good to merge!

**Pull request merged**